src/Tables.jl

```diff
@@ -85,11 +85,16 @@ abstract type Table end
 abstract type AccessStyle end
 struct RowAccess <: AccessStyle end
 struct ColumnAccess <: AccessStyle end
-AccessStyle(x) = RowAccess()
 
 "Tables.schema(s) => NamedTuple{names, types}"
 function schema end
-schema(x) = eltype(x)
+
+function istable(x::T) where {T}
+    hasmethod(AccessStyle, Tuple{T}) &&
+    hasmethod(schema, Tuple{T}) &&
+    AccessStyle(T) === RowAccess() ? hasmethod(rows, Tuple{T}) :
```

Comment:
Very nice! I have only the question if we need the tests `hasmethod(rows, Tuple{T})` in this line and `hasmethod(columns, Tuple{T})` in line below. I guess they will always evaluate to `true` as `rows` and `columns` functions accept any `x` in definitions below.